server/routes/NodeIndexRoute.js

```diff
@@ -6,7 +6,35 @@
 const express = require('express');
 const nlpService = require("../services/NlpService");
 const dataService = require("../services/DataService");
-const router = express.Router();
+const crypto = require('crypto');
+var router = express.Router();
+var qs = require('querystring');
+var multer = require('multer');
+var path = require('path');
+var fs = require('fs');
+
+var DIR = "./server/data/";
+var storage = multer.diskStorage({
+  destination: DIR,
+  filename: function (req, file, cb) {
+    crypto.pseudoRandomBytes(16, function (err, raw) {
+      if (err) return cb(err);
+      fs.exists(DIR + file.originalname, function(exists) {
+        var fileName;
+        if (exists) {
+          console.log("exists");
+          fileName = Date.now() + '_' + file.originalname;
+        } else {
+          console.log("does not exist");
+          fileName = file.originalname;
+        } 
+        cb(null, fileName)
+    });
+    })
+  }
+});
+
+var upload = multer({storage: storage}).single('fileItem');
```

Comment:
I would advise to move all this in the dataservice and keep the router cleaner

Reply:
I moved code to dataservice.js